src/hazelcore/Builtins.re

```diff
@@ -0,0 +1,66 @@
+/*
+   Built-in functions for Hazel.
+
+   To add a built-in function or constant, write the implementation in the
+   `Impls` module below and add it to the `builtins` list.
+
+   See the existing ones for reference.
+ */
+
+module Impls = {
+  open EvaluatorResult;
+
+  /* int_of_float implementation. */
+  let int_of_float = (ident, r1) =>
+    switch (r1) {
+    | BoxedValue(FloatLit(f)) =>
+      let i = int_of_float(f);
+      BoxedValue(IntLit(i));
+    | BoxedValue(d1)
```

Comment:
if its any boxed value other than a float lit it should raise an exception not return Indet (also in other cases in this file)